sdk/python/kfp/components/_component_store.py

```diff
@@ -154,28 +158,43 @@ def _load_component_spec_in_component_ref(
                     str(component_path))
                 return component_ref
 
+        #Trying URI template
+        if self.uri_search_template:
+            url = self.uri_search_template.expand(
+                name=name, digest=digest, tag=tag)
+            tried_locations.append(url)
+            if self._load_component_spec_from_url(component_ref, url):
+                return component_ref
+
         #Trying URL prefixes
         for url_search_prefix in self.url_search_prefixes:
             url = url_search_prefix + path_suffix
             tried_locations.append(url)
-            try:
-                response = requests.get(
-                    url, auth=self._auth
-                )  #Does not throw exceptions on bad status, but throws on dead domains and malformed URLs. Should we log those cases?
-                response.raise_for_status()
-            except:
-                continue
-            if response.content:
-                # TODO: Verify that the content matches the digest (if specified).
-                component_ref.url = url
-                component_ref.spec = comp._load_component_spec_from_yaml_or_zip_bytes(
-                    response.content)
+            if self._load_component_spec_from_url(component_ref, url):
                 return component_ref
 
         raise RuntimeError(
             'Component {} was not found. Tried the following locations:\n{}'
             .format(name, '\n'.join(tried_locations)))
 
+    def _load_component_spec_from_url(self, component_ref, url):
```

Comment:
nit: worth having a docstring for this as well

Reply:
Will do. This code still isn't as DRY as I'd like, iirc this extracted method (or something very similar) appears elsewhere in the code too.

I see some pylint warnings on this file too, but I don't think they're a result of the small changes I've made. For example:

```
Instance of 'ModelBase' has no 'implementation' memberpylint(no-member)
```

Reply:
In terms of my DRY comments: there's a lot of similarity between this method and those in `_components.py`, but as I was extracting code that already existed in the component store I tried to avoid further refactoring.